### ISSUE TITLE
cmd_places_showInFinder returns an error

### ISSUE THREAD

**Issue by Defman21** [Komodo/KomodoEdit#250]
Reproduced on Linux DE's:
* Pantheon (File manager: `pantheon-files 0.2.0`)
* Mate (File manager: `Caja`)

The error:
```
[Exception... "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [koISysUtils.ShowFileInFileManager]"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: chrome://komodo-places/content/places.js :: ManagerClass.prototype.doShowInFinder :: line 2372"  data: no]
```

**Comment by cgchoffman** [comment]
https://github.com/Komodo/KomodoEdit/blob/9.0.1/src/components/koSysUtils.py#L247
Perhaps swapped to querying xdg-mime
```xdg-mime query default inode/directory```  

https://wiki.archlinux.org/index.php/Xdg-open

**Comment by Defman21** [comment]
```python
    def ShowFileInFileManager(self, filename):
        # nsILocalFile handles some of this
        localFile = components.classes["@mozilla.org/file/local;1"].createInstance(components.interfaces.nsILocalFile)
        localFile.initWithPath(filename)
        try:
            localFile.reveal()
            return
        except COMException, e:
            # reveal is not implemented, try the old stuff
            pass
        if sys.platform not in ['win32','darwin']:
            manager = self._getManager()
            if os.path.isfile(filename):
                filename = os.path.dirname(filename)
            os.system('xdg-open "%s" &' % filename)
```
Works fine!

**Comment by cgchoffman** [comment]
What kind of file did you try it with?  That method doesn't work for HTML, XML, or I assume, other files that have a different default program that they open in.

**Comment by Defman21** [comment]
I tried it only with folders and forgot about files... huh.

**Comment by Defman21** [comment]
Well, I've tried it with html, php, css and xml files. All of them works fine. If I click on a html file *IN* files manager - Firefox open the file.

**Comment by cgchoffman** [comment]
Sounds like the behavior isn't consistent then.  xdg-meme sounds like a more robust method to implement here.

**Comment by Defman21** [comment]
Maybe you didn't understand me properly...
https://youtu.be/2RrXcWnyVQs

**Comment by cgchoffman** [comment]
Yes I got it @Defman21.  As I said already, I did not get the same functionality when using that tool.  That is what I mean by "the behavior isn't consistent".  So it will likely not work for us here or else different people will get a different result when using the same tool on a different OS.

**Comment by Defman21** [comment]
I'll dig in it as far as I can to see what can I do to fix that :smiley:

**Comment by ervumlens** [comment]
Regarding `xdg-open` vs. `xdg-mime` (and assuming this issue hasn't been addressed behind the scenes already)...

Reading back through the comments, it's not clear whether the proposal was to call `xdg-open /path/to/selected.file` or `xdg-open /path/to/`. @cgchoffman  mentions "That method [?] doesn't work for HTML, XML, or I assume, other files that have a different default program that they open in". This implies passing the target file to `xdg-open`. Passing only the target file's directory name to `xdg-open`, as shown in @Defman21 's code example, will never open the target file's associated editor because only the directory name is passed to it. (The current code passes only the directory name to `nautilus` or `konqueror`.)

I don't know if exactly this was proposed, but I'd be wary of passing a non-directory name to the application mapped in `xdg-mime query default inode/directory` simply because it's only known to handle directories correctly. The mapped application may open regular files in their respective editors, depending on the particular distro/window manager/user configuration.

On a side note, the world's laziest workaround for this issue is to create a shell script named `nautilus` that calls `xdg-open $*`. :neckbeard:

**Comment by ervumlens** [comment]
FWIW, Atom [had the same problem](https://github.com/atom/tree-view/pull/144/files) and landed on using `xdg-open /path/to/`. The current version of the linked file is harder to follow now, but it continues to use `xdg-open`.

**Comment by Defman21** [comment]
Ervumlens gave a good point so passing the directory name to xdg-open should be fine. Should I send a PR for that?

**Comment by Naatan** [comment]
Feel free! :)

**Comment by Defman21** [comment]
Yay! :^)
